### PR TITLE
Add "step":"ERROR" to ILM explain response for missing policy

### LINKED ISSUE #99030: [ILM] Always show step if show step_info
### Description

👋🏼 howdy, team! 

I've noticed when troubleshooting ILM that I sometimes miss (very obvious user error) problems because I filter by `step:ERROR`. However, it appears if you create a test index with bogus ILM policy, you can return the step's error under `step_info` BUT not get `step:ERROR` to filter on

## setup
```
PUT test
{
    "settings": {
        "lifecycle": {
            "rollover_alias": "oops",
            "name": "also_oops"
        }
    }
}
```
(wait for ILM to poll at least once)

## results

```diff
> GET test/_ilm/explain?only_errors=true
{
  "indices": {
    "test": {
      "policy": "also_oops",
      "index": "test",
-    # no step:ERROR
      "managed": true,
      "step_info": {
        "reason": "policy [also_oops] does not exist",
        "type": "illegal_argument_exception"
      }
    }
  }
}
```

## effect

The ILM Explain with `only_errors=true` (above) does NOT agree with the Kibana UI which filters `ilm.step:ERROR` which causes indices requiring manual intervention to fall between the cracks when they shouldn't.

image

image

### PR DESCRIPTION
**Pull request by mattc58**
This adds the `"step":"ERROR"` field in the ILM explain response for the case when there's a missing policy.

It changes the `_ilm/explain` response to look like:
```
{
    "indices": {
        "test2": {
            "index": "test2",
            "index_creation_date_millis": 1703698545568,
            "managed": true,
            "policy": "also_oops",
            "step": "ERROR",
            "step_info": {
                "reason": "policy [also_oops] does not exist",
                "type": "illegal_argument_exception"
            },
            "time_since_index_creation": "21.44s"
        }
    }
}
```

Closes #99030

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/IndexLifecycleExplainResponse.java [hunk 1]
```diff
@@ -227,9 +227,22 @@ private IndexLifecycleExplainResponse(
...
+            if (ErrorStep.NAME.equals(step)) {
+                maxNull = 2;
+                conditions = Stream.of(phase, action);
+            } else {
+                maxNull = 3;
+                conditions = Stream.of(phase, action, step);
+            }
+
+            long numNull = conditions.filter(Objects::isNull).count();
+            if (numNull > 0 && numNull < maxNull) {
```
Comment (dakrone) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/IndexLifecycleExplainResponse.java:245]: I think this method might now be too clever; it's hard to reason about. Can we make it simpler? Perhaps something like:

```java
// If at least one detail is null, but not *all* are null
if (Stream.of(phase, action, step).anyMatch(Objects::isNull) && Stream.of(phase, action, step).allMatch(Objects::isNull) == false) {
    // …and it's not in the error step
    if (ErrorStep.NAME.equals(step) == false) {
        // then throw an exception about missing the complete step details
        throw new IllegalArgumentException("…");
    }
}
```

File: x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportExplainLifecycleAction.java [hunk 2]
```diff
@@ -152,7 +152,7 @@ static IndexLifecycleExplainResponse getIndexLifecycleExplainResponse(
                     originationDate != -1L ? originationDate : lifecycleState.lifecycleDate(),
                     lifecycleState.phase(),
                     lifecycleState.action(),
-                    lifecycleState.step(),
+                    indexLifecycleService.policyExists(policyName) == false ? ErrorStep.NAME : lifecycleState.step(),
```
Comment (dakrone) [inline x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportExplainLifecycleAction.java:155]: Can you add a comment above this line explaining why we ingest the "error" name here?

**Review by dakrone — COMMENTED** [review]
I left a couple of minor comments

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/IndexLifecycleExplainResponse.java (hunk 1 above)
Comment (mattc58) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/IndexLifecycleExplainResponse.java]: Way more clear--thanks!

**Review by dakrone — APPROVED** [review]
LGTM

**Pull request merged**